mbq/ranch/killswitch.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from typing import Callable
+
+from django.conf import settings
+
+import celery
+from mbq import metrics
+
+
+logger = logging.getLogger(__name__)
+
+
+_collector = metrics.Collector(namespace="mbq.ranch")
+
+
+def create_killswitch_task_class(variation: Callable[[str, bool], bool]) -> celery.Task:
+    class KillSwitchTask(celery.Task):
+        def killswitch_name(self):
+            service = settings.RANCH["service"]
+            return f"task-killswitch-{service}-{self.name}".replace(".", "-").replace(
+                "_", "-"
+            )
+
+        def __call__(self, *args, **kwargs):
+            if variation(self.killswitch_name(), False):
+                from . import _collector
```

Comment:
Why do we import here and at the top of the module?

Reply:
because this import can't happen until django is initialized. The collector comes from app.ready

Reply:
And the collector comes from app.ready because it uses tags from settings.RANCH

Reply:
fixed